starter-code/index.js

```diff
@@ -1,10 +1,89 @@
-var SortedList = function() {};
-
-SortedList.prototype.add     = function(item) {}
-SortedList.prototype.get     = function(pos) {}
-SortedList.prototype.max     = function() {}
-SortedList.prototype.min     = function() {}
-SortedList.prototype.average = function() {}
-SortedList.prototype.sum     = function() {}
+let SortedList = function() {
+    this.items = [];
+    this.length = this.items.length;
+};
+
+SortedList.prototype.add = function(item) {
+    
+    this.items.push(item)
+    this.items.sort(function(i1, i2){
+        return i1-i2});
+    
+    this.length++;
+}
+SortedList.prototype.get = function(pos) {
+    if(pos > this.items.length){
+        return this.OutOfBounds();
+    }
+    return this.items[pos-1];
+}
+SortedList.prototype.max = function() {
+    if (this.items === 0) {
+        return this.EmptyList();
+    } else {
+        var max = this.items[0];
+        this.items.forEach(function(e) {
+            if(e > max){
+                max = e;
+            }
+        }.bind(this));
+    }
+    return max; 
+}
+SortedList.prototype.min = function() {
+    if (this.items.length === 0) {
+        return this.EmptyList();
+    } else {
+        var min = this.items[0];
+        this.items.forEach(function(e) {
+            if(e < min){
+                min = e;
+            }
+        }.bind(this));
+    }
+    return min;
+}
+SortedList.prototype.average = function() {
+    if (this.items.length === 0) {
+        return this.EmptySortedList();
+    } else {
+        return this.items.reduce(function(avg, e, i, arr) {
+            var average = 0;
+            if (i === this.items.length - 1) {
+                return average = ((avg + e) / this.items.length);
+            } else {
+                return average = avg + e;
+            }
+        }.bind(this), 0);
+        return average;
+    }
+}
+
+
+SortedList.prototype.sum = function() {
+    if (this.items.length === 0) {
+        return this.EmptySortedList();
+    } else {
+        return this.items.reduce(function(sum, e) {
+          return sum += e;
```

Comment:
Aquí igual que antes, por qué `return sum += e` si es suficiente con `return sum + e` cuando utilizáis `reduce()`